core/trino-main/src/test/java/io/trino/split/TestBufferingSplitSource.java

```diff
@@ -35,7 +37,14 @@
 
 public class TestBufferingSplitSource
 {
-    private static final Executor executor = new BoundedExecutor(newCachedThreadPool(daemonThreadsNamed(TestBufferingSplitSource.class.getSimpleName() + "-%s")), 10);
+    private final ExecutorService executorService = newCachedThreadPool(daemonThreadsNamed(TestBufferingSplitSource.class.getSimpleName() + "-%s"));
+    private final Executor executor = new BoundedExecutor(executorService, 10);
+
+    @AfterClass(alwaysRun = true)
```

Comment:
This needs to be JUnit's `@AfterAll`. Also, you're going to have to add this as a class annotation:

```
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
@Execution(ExecutionMode.CONCURRENT)
```

Reply:
Right now this test class is using TestNG annotations and not JUnit ones- does that matter here or is there some JUnit / TestNG bridging going on?

Reply:
Woops, I'm wrong about that- seems this was updated. I'll switch to the JUnit annotations